packages/interactor/types/locators.ts

```diff
@@ -0,0 +1,43 @@
+import { createInteractor } from '../src/index';
+
+const Link = createInteractor<HTMLLinkElement>('link')({
+  selector: 'a',
+  defaultLocator: ['byHref', 'byTitle'],
+  locators: {
+    byHref: (element) => element.href,
+    byTitle: (element) => element.title
+  }
+});
+
+Link('foobar');
+
+Link.byHref('foobar');
+
+// cannot use wrong type argument on locator
+// $ExpectError
+Link.byHref(123);
+
+// cannot use locator which is not defined
+// $ExpectError
+Link.byValue('foobar');
```

Comment:
@dagda1 This is also not erroring, but I feel like once one of these problems is resolved the other will be resolved too.

Reply:
This should fix it

```typescript
export type LocatorImplementation<E extends Element, S extends InteractorSpecification<E, L>, L extends LocatorSpecification<E>> = {
  [P in keyof S['locators']]: S['locators'][P] extends ((element: E, ...args: unknown[]) => unknown) ? (value: string, filters?: FilterImplementation<E, S>) => InteractorInstance<E, S, L> : never;
}
```

Reply:
Of note is that this is not erroring as expected only because of my modifications in attempt to get the other expected error working.